### PR TITLE
fix: upd build cache documentation

### PR DESCRIPTION
**Pull request by parthasl**
## Objective

Add build-cache documentation on how to handle cache updates from parallel jobs

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/configuration/build-cache.md [hunk 1]
```diff
@@ -96,6 +96,17 @@ jobs:
         cache: false
 ```
 
+## Update Cache from parallel jobs
+
+Please be cautious when specifying same cache path at an event or pipeline level in screwdriver.yaml having parallel jobs. 
+When jobs run in parallel there is no guarantee which job will succeed first in writing cache, therefore the same cache path 
+might get overwritten or lost.
+
+How to handle above scenario?
```
Comment (tkyi) [inline docs/user-guide/configuration/build-cache.md]:
```suggestion
How to handle the above scenario? 
```

File: docs/user-guide/configuration/build-cache.md [hunk 2]
```diff
@@ -96,6 +96,17 @@ jobs:
         cache: false
 ```
 
+## Update Cache from parallel jobs
```
Comment (tkyi) [inline docs/user-guide/configuration/build-cache.md]:
```suggestion
## Update Cache from Parallel Jobs
```

**Review by tkyi — COMMENTED** [review]
Also need to add to TOC

**Comment by parthasl** [comment]
> Also need to add to TOC

done

**Review by tkyi — APPROVED** [review]

**Pull request merged**